### PR TITLE
chore(sdk): Added logic to render EmptyComponent in both SDK

### LINKED ISSUE #28728: Angular SDK: Manage "NoComponent" on SDK
### Parent Issue

https://github.com/dotCMS/core/issues/28404

### Task

Create a workaround the "NoComponent" logic.

- [ ] In Live mode, the "NoComponent" should not be rendered
- [ ] In Dev mode, the "NoComponent" should be rendered, and this component should be loaded by developer. Like
```
{
banner: BannerComponent,
...,
empty: CustomEmptyComponent
}
```

- [ ] In case the developer do not configure a custom no component, show the current no component from the library

### Proposed Objective

Same as Parent Issue

### Proposed Priority

Priority 3 - Average

### Acceptance Criteria

_No response_

### External Links... Slack Conversations, Support Tickets, Figma Designs, etc.

_No response_

### Assumptions & Initiation Needs

_No response_

### Quality Assurance Notes & Workarounds

_No response_

### Sub-Tasks & Estimates

_No response_

### PR DESCRIPTION
**Pull request by KevinDavilaDotCMS**
### Proposed Changes
* Added Custom EmptyComponent
* Now the EmptyComponent is rendered 

Angular SDK

https://github.com/dotCMS/core/assets/144152756/ccd4c690-b8e8-4fa8-94ae-1c0569db0ecd

React SDK

https://github.com/dotCMS/core/assets/144152756/003701ed-7a40-4275-b9bf-fabaee9f6ccf

### CODE REVIEW

**Review by zJaaal — APPROVED** [review]

**Review by oidacra — APPROVED** [review]

File: core-web/libs/sdk/angular/src/lib/layout/container/container.component.html [hunk 1]
```diff
@@ -4,14 +4,12 @@
 <dotcms-contentlet-wrapper [contentlet]="contentlet" [container]="$dotContainerAsString()">
     <ng-container
         *ngComponentOutlet="
-            (componentsMap[contentlet.contentType] | async) || NoComponent;
+            (componentsMap[contentlet.contentType] || componentsMap['Empty'] | async) ||
```
Comment (fmontes) [inline core-web/libs/sdk/angular/src/lib/layout/container/container.component.html:7]: POWER!

File: core-web/libs/sdk/react/src/lib/components/Container/Container.tsx [hunk 2]
```diff
@@ -49,7 +53,12 @@ export function Container({ containerRef }: ContainerProps) {
           };
 
     const ContainerChildren = contentlets.map((contentlet) => {
-        const Component = components[contentlet.contentType] || NoContent;
+        const ContentTypeComponent = components[contentlet.contentType];
+        const DefaultComponent = components['Empty'] || NoComponent;
```
Comment (fmontes) [inline core-web/libs/sdk/react/src/lib/components/Container/Container.tsx:57]:
```suggestion
        const DefaultComponent = components['XXXX'] || NoComponent;
```

Key `Empty` doesn't represent the intention here.

File: examples/angular/src/app/pages/content-types/empty/empty.component.ts [hunk 3]
```diff
@@ -0,0 +1,9 @@
+import { Component } from '@angular/core';
+
+@Component({
+  selector: 'app-empty',
+  standalone: true,
+  template: ` <div class="relative w-full bg-gray-200 h-12 flex justify-center items-center overflow-hidden">Dont have a component for this contentlet.</div> `,
```
Comment (fmontes) [inline examples/angular/src/app/pages/content-types/empty/empty.component.ts:6]:
```suggestion
  template: ` <div class="relative w-full bg-gray-200 h-12 flex justify-center items-center overflow-hidden">No component for this contentlet or widget.</div> `,
```

File: examples/nextjs/src/components/my-page.js [hunk 4]
```diff
@@ -18,53 +18,55 @@ import { withExperiments } from "@dotcms/experiments";
...
-  Banner: Banner,
-  Activity: Activity,
-  Product: Product,
-  Image: ImageComponent
+    webPageContent: WebPageContent,
+    Banner: Banner,
+    Activity: Activity,
+    Product: Product,
+    Image: ImageComponent,
+    Empty: () => <div class="relative w-full bg-gray-200 h-12 flex justify-center items-center overflow-hidden">Dont have a component for this contentlet.</div>,
```
Comment (fmontes) [inline examples/nextjs/src/components/my-page.js:33]: Keep this consistent with the external file.

Comment (fmontes) [reply examples/nextjs/src/components/my-page.js:33]: Well... maybe is cool to show that we can do inline stuff.

But also is important to show the content type name here.

**Review by fmontes — CHANGES_REQUESTED** [review]

File: core-web/libs/sdk/react/src/lib/components/Container/Container.tsx (hunk 2 above)
Comment (KevinDavilaDotCMS) [reply core-web/libs/sdk/react/src/lib/components/Container/Container.tsx:57]: maybe `CustomNoComponent` ?

File: examples/nextjs/src/components/my-page.js (hunk 4 above)
Comment (KevinDavilaDotCMS) [reply examples/nextjs/src/components/my-page.js]: Changed!

File: examples/angular/src/environments/environment.development.ts [hunk 5]
```diff
@@ -1,5 +1,7 @@
-export const environment = {
-    dotcmsUrl: "http://localhost:8080",
-    authToken: "eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJzdWIiOiJhcGlkZTU4MDZmNS1iZDZlLTRkMTYtOWVhZi1jM2ZmMTRiYzIwZmUiLCJ4bW9kIjoxNzE3NjE4MDAwMDAwLCJuYmYiOjE3MTc2MTgwMDAsImlzcyI6IjhmOGJjZWUxZTUiLCJsYWJlbCI6ImtkIiwiZXhwIjoxODEyMTcxNjAwLCJpYXQiOjE3MTc2MTgwMDAsImp0aSI6IjNiNDI1NjQ5LTNlNDUtNGM2My1iOTE3LWE5MGUwMGM4M2UxYiJ9.niN9ZWPGxit5JuL94t89gNPpVSxWV--9OkJpwGjInZQ",
-    siteId: "true",
+const environment = {
+  dotcmsUrl: 'http://localhost:8080',
+  // This must be a read-only token.
```
Comment (fmontes) [inline examples/angular/src/environments/environment.development.ts:3]: "This auth token will be exposed in the client-side requests, for security purposes, make sure is read-only. More info: https://www.dotcms.com/docs/latest/permissions"

**Review by fmontes — CHANGES_REQUESTED** [review]

**Review by fmontes — APPROVED** [review]

**Comment by rjvelazco** [comment]
### Passed Internal QA

- Tested on docker: [`dotcm/dotcms:trunk_ 7b54550`]

#### Video

https://github.com/dotCMS/core/assets/72418962/d30e759a-88c8-43ca-af1f-450c76436ed6

**Pull request merged**